src/ripple/app/tx/impl/Escrow.cpp

```diff
@@ -156,14 +111,26 @@ EscrowCreate::preflight (PreflightContext const& ctx)
     if (ctx.tx[sfAmount] <= beast::zero)
         return temBAD_AMOUNT;
 
-    if (! ctx.tx[~sfCancelAfter] &&
-            ! ctx.tx[~sfFinishAfter])
-        return temBAD_EXPIRATION;
+    // We must specify at least one timeout value
+    if (! ctx.tx[~sfCancelAfter] && ! ctx.tx[~sfFinishAfter])
+            return temBAD_EXPIRATION;
 
+    // If both finish and cancel times are specified then the cancel time must
+    // be strictly after the finish time.
```

Comment:
By requiring cancel after to be strictly after finish after, we need to check if cancel after < finish after in `doApply` instead of `preflight` because we need to check the switch. `fix1571` is also requiring that the `finishAfter` and `cancelAfter` times are strictly greater than the close time (before it was greater or equal). This also requires code changes to `doApply`.

If this were brand new code, I would agree with this semantics. However, requiring that finish after is one second more than cancel after is essentially the old semantic of `<=`.

What do you think about either:
1) Keeping the old `<=` semantic and reducing the number of changes in this PR.
or
2) Have a more meaningful time that cancel after must be greater than  finish after. One minute? Thirty seconds?